utils/notification.ts

```diff
@@ -142,6 +142,37 @@ const successNH = (): NHResultSuccess =>
     kind: "SUCCESS"
   });
 
+const dictionaryIsEmpty = (
+  obj: Azure.ServiceBus.Dictionary<string | unknown>
+): boolean => Object.keys(obj).length === 0;
+
+const handleResponseOrError = (
+  resolve: (
+    value:
+      | { readonly kind: "SUCCESS" }
+      | PromiseLike<{ readonly kind: "SUCCESS" }>
+  ) => void,
+  reject: (reason?: unknown) => void
+) => (err: Error | null, response: Azure.ServiceBus.Response): void =>
```

Comment:
Code design: 
instead of using callback, it's preferable to have the function to return a value (example: `Either<Bad, Good>`. 
Benefits:
* explicit signature, it's easier to understand what to expect from this function
* you let the user decide how to compose it

As a general advice, try to delegate type-lifting to the caller as it leads to more composability

Reply:
Another possibility is this function to have a `PromiseLike` signature (which is more or less what it has already), so that it can be _piped_ with other promises (as promises are meant to be).

Example:
```ts
      new Promise<NHResultSuccess>((resolve, reject) =>
        notificationHubService.createOrUpdateInstallation(
          azureInstallationOptions,
          (err, res) => handleResponseOrError(err, res).then(resolve).catch(reject)
        )
      ),
```
Or even:
```ts
      promisify(notificationHubService.createOrUpdateInstallation)(
          azureInstallationOptions
      ).catch(err => /* just the unhappy path of handleResponseOrError*/)
```

Reply:
I get it. In this case the original callback is required by the notificationHubService, and we use the reject because of the wrapping `tryCatch`, if I undestand it correctly.
Anyway, I tried to follow your advices (by doing this we are braking eslint rules sonarjs/no-identical-functions)
Let me know what do you think about this new implementation.
Thanks